packages/xforms-engine/src/client/text/RichTextUnit.ts

```diff
@@ -0,0 +1,112 @@
+ * least for now and the foreseeable future, we **DO NOT** expect to be able to
+ * use a {@link RichTextUnit} or any of its subtypes as, say, XPath evaluation
+ * context nodes. In fact, they will frequently be a _product of XPath
+ * evaluations_! So some caution is warrranted about whether a name like
+ * `RichTextNode` might inspire unintended assumptions.
+ *
+ * There is also a social (team/contributor) caution: do **we** feel comfortable
+ * tossing another "node" concept into the proverbial "node soup"? Will **we**
+ * understand what it means (and more importantly _doesn't mean_) in the coming
+ * months... or years?
+ *
+ * If we have consensus on these cautions:
+ *
+ * 1. Is "unit" a reasonable substitution for what is naturally a "node"?
+ * 2. Would we prefer to make other naming revisions to further distance the
+ *    concept from "node"-ness? I.e. should we consider another name for
+ *    {@link children}? For example: in an earlier iteration of this design
+ *    proposal, the property was named `spans`—which recognized that any "unit
+ *    with children" (née "parent node") would only ever contain "span"
+ *    children.
+ */
+export interface RichTextUnit {
+	readonly type: RichTextUnitType;
+
+	/**
+	 * Represents variants of a given {@link RichTextUnitType | unit type}, to be
+	 * defined and narrowed within the subtypes of each.
+	 */
+	readonly variant: unknown;
+
+	get children(): readonly RichTextUnit[];
+
+	/**
+	 * Represents the text value—as parsed and/or in its present computed
+	 * state—with any formatting syntax removed or reprocessed; i.e. producing a
+	 * string suitable for presentation to a user where "rich" or "formatted" text
+	 * are unsuitable.
+	 *
+	 * **WARNING:** This value may be computed from several sources (e.g.
+	 * `jr:itext()`, `<output>`, inline text), and may be recomputed more
+	 * frequently than its {@link children | structured counterpart}.
+	 */
+	get plainText(): string;
+
+	/**
+	 * Represents the raw source text—as parsed and/or in its present computed
+	 * state—without any further processing.
+	 *
+	 * **WARNING:**
+	 *
+	 * - The raw text value is exposed for non-presentational use cases (e.g.
+	 *   testing). Clients wishing to present text values _without formatting_ are
+	 *   strongly encouraged to reference {@link plainText} instead.
+	 *
+	 * - This value may be computed from several sources (e.g. `jr:itext()`,
+	 *   `<output>`, inline text), and may be recomputed more frequently than its
+	 *   {@link children | structured counterpart}.
+	 *
+	 *
+	 * @todo [Design review detail, remove before implementation.]
+	 *
+	 * Do we actually want to expose this? The most likely use case would be
+	 * testing. The risk of exposing it is that it could encourage conflicting or
+	 * divergent implementations of ODK's flavor of Markdown (diverging in terms
+	 * of parsing, semantics, capabilities, and so on).
```

Comment:
_Actually_, if we lean more toward an approach where **clients determine** which `RichText` cases have what capabilities, access to `rawText` may be important for that! For example, it would allow a client receiving the same `RichText` values for `label` and `hint` to render:

- `<label>**foo** bar</label>` -> `<label><strong>foo</strong bar</label>`
- `<hint>**foo** bar</hint>` -> `<p>**foo** bar</p>`

With the same parsing logic, and without access to `rawText`, those different renderings would probably not be possible.